### PR TITLE
[OSDEV-1453] Use the 'detail' keyword instead of 'message' in response objects

### PR DESCRIPTION
**Pull request by Innavin369**
[OSDEV-1453](https://opensupplyhub.atlassian.net/browse/OSDEV-1453)  Use the detail keyword instead of message in response objects for endpoints in V1 specs.

Screenshot 2024-11-28 at 11 43 58

[OSDEV-1453]: https://opensupplyhub.atlassian.net/browse/OSDEV-1453?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Screenshot 2024-11-28 at 13 49 44

### CODE REVIEW

File: src/django/api/serializers/v1/opensearch_common_validators/status_validator.py [hunk 1]
```diff
@@ -16,14 +16,14 @@ def validate_opensearch_params(self, data) -> List[dict]:
...
         elif not all(item in self.VALID_STATUSES for item in status):
             invalid_statuses = [item for item in status if
                                 item not in self.VALID_STATUSES]
             errors.append({
                 "field": "status",
-                "message": (
+                "detail": (
                     f"Invalid source(s) {invalid_statuses}. "
                     "Allowed values are 'PENDING','APPROVED' or 'REJECTED'."
                 )
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/django/api/serializers/v1/opensearch_common_validators/status_validator.py:29]: changed

File: src/django/api/tests/test_production_locations_view_set.py [hunk 2]
```diff
@@ -42,6 +42,6 @@ def test_get_single_production_location_not_found(self):
         self.search_index_mock.return_value = {}
         url = f"/api/v1/production-locations/{self.os_id}/"
         api_res = self.client.get(url)
-        message = {"detail": "The location with the given id was not found."}
-        self.assertEqual(api_res.data, message)
+        detail = {"detail": "The location with the given id was not found."}
+        self.assertEqual(api_res.data, detail)
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/django/api/tests/test_production_locations_view_set.py:46]: these files do not regard v1

File: src/django/api/tests/test_facility_claim_view_set.py [hunk 3]
```diff
@@ -206,7 +206,7 @@ def test_message_claimant_no_message(self):
         )
 
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
-        self.assertEqual(response.data['detail'], 'Message is required.')
+        self.assertEqual(response.data['message'], 'Message is required.')
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/django/api/tests/test_facility_claim_view_set.py]: this file do not regard to v1

File: src/django/api/tests/test_facility_claim_view_set.py [hunk 4]
```diff
@@ -188,7 +188,7 @@ def test_message_claimant_permission_denied(self):
 
         self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
         self.assertEqual(
-            response.data['detail'],
+            response.data['message'],
             'Authentication credentials were not provided.',
         )
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/django/api/tests/test_facility_claim_view_set.py]: this file do not regard to v1

File: src/django/api/tests/test_v1_utils.py [hunk 5]
```diff
@@ -130,18 +130,18 @@ def test_handle_value_error(self):
...
+            response.data['errors'][0]['detail']
         )
 
     def test_handle_opensearch_exception(self):
         exception = OpenSearchServiceException("OpenSearch error")
         response = handle_opensearch_exception(exception)
         self.assertIsInstance(response, Response)
         self.assertEqual(response.status_code, 500)
-        self.assertEqual(response.data['message'], "OpenSearch error")
+        self.assertEqual(response.data['detail'], "OpenSearch error")
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/django/api/tests/test_v1_utils.py:147]: these changes do not regard the task

File: src/django/api/views/v1/utils.py [hunk 6]
```diff
@@ -92,7 +92,7 @@ def handle_opensearch_exception(e):
     logger.error(f'Internal server error: {e}')
     return Response(
         {
-            "message": f"{e}"
+            "detail": f"{e}"
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply src/django/api/views/v1/utils.py:95]: this change does not regard v1

**Review by VadimKovalenkoSNF — APPROVED** [review]
Approved.

**Pull request merged**